updater.go

```diff
@@ -244,19 +245,108 @@ func (u *Updater) AppendHeaderAt(fh *FileHeader, offset int64) (io.Writer, error
 		return nil, err
 	}
 
-	if offset < 0 {
-		offset = u.dirOffset
-	}
 	// Offset should match existing header offsets or equal to directory offset.
 	var offsetExists bool
-	for i, h := range u.dir {
-		if h.offset == uint64(offset) {
-			offsetExists = true
-			// Delete the corresponding header.
-			u.dir = append(u.dir[:i], u.dir[i+1:]...)
-			break
+
+	existingFileIndex := 0
```

Comment:
Should the `existingFileIndex` be initialized to `-1`? Since the file may be at the top of the zip file, and its index is counted from 0.

Reply:
You are absolutely right!